src/js/controller/events-middleware.ts

```diff
@@ -1,13 +1,15 @@
-export default function middleware(model, type, currentState) {
-    let newState = currentState;
+import { GenericObject } from 'types/generic.type';
+
+export default function middleware(model: GenericObject, type: string, currentState: GenericObject): GenericObject {
```

Comment:
Should `model` be the `CoreModel` type here?

Reply:
Awesome catch 👍